modules/cudawarping/src/warp.cpp

```diff
@@ -193,7 +260,8 @@ void cv::cuda::warpAffine(InputArray _src, OutputArray _dst, InputArray _M, Size
     CV_Assert( interpolation == INTER_NEAREST || interpolation == INTER_LINEAR || interpolation == INTER_CUBIC );
     CV_Assert( borderMode == BORDER_REFLECT101 || borderMode == BORDER_REPLICATE || borderMode == BORDER_CONSTANT || borderMode == BORDER_REFLECT || borderMode == BORDER_WRAP );
 
-    _dst.create(dsize, src.type());
+    if (_dst.size() != dsize)
+        _dst.create(dsize, src.type());
```

Comment:
I think this is redundant as [void cv::cuda::GpuMat::create(int _rows, int _cols, int _type)](https://github.com/opencv/opencv/blob/c2c8da2517f57eae3d3db459ab36c3813692d87e/modules/core/src/cuda/gpu_mat.cu#L160) should automatically handle this for you in addition to checking the type.

Reply:
far as I can remember, I checked the profiler and maybe there was some overhead? I am not sure...